common/djangoapps/pipeline_mako/templates/static_content.html

```diff
@@ -199,6 +199,13 @@
     return get_value('SHOW_LANGUAGE_SELECTOR', settings.FEATURES.get('SHOW_LANGUAGE_SELECTOR', False))
 %></%def>
 
+<%def name="show_footer_language_selector(api_setting)"><%
```

Comment:
I didn't realize that ``show_language_selector`` was already in this file. This is an inappropriate place for that method too, IMO! I also don't like that the two methods have different signatures, but I suppose that's because we don't render the header via an API.

I feel that this method is too confusing because it is serving different purposes, and the contract has to understand them. How would you write a unit test for this method? Why should the template have to know whether it is being rendered by an API or not? The API only renders in one method, so it seems cleaner to have that pass an extra parameter, rather than having this method need to understand about the API's requirements.

This is one of the unfortunate things about Mako, which is that it lets you evaluate arbitrary Python, so the code gets much more confusing. If this was a Django Template, it would receive the context parameter and just render based upon it. It wouldn't have to know about Django settings or API values.

I wonder if it would be cleaner to have this defaulting logic applied here:

https://github.com/edx/edx-platform/blob/master/common/djangoapps/edxmako/shortcuts.py#L138

Then the templates could rely upon this parameter always being passed through.

Sorry for the philosophical discussion. Let me know if any of this makes sense. Maybe I should just let this go since this code is already confusing, but I'd rather not add to the complexity if we can avoid it.